Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.36320&attempt_id=4b08e0d7-f6a4-421c-a3d5-f77e2a886936)

<details><summary>Partial log (click to expand)</summary><p>

```
export LD_PRELOAD=../src/libgsettings-qt.so.1; qmlplugindump -notrelocatable GSettings 1.0 .. > //nix/store/iwcbfv0p110qkqxn6wjmm3kz2zs2140m-gsettings-qt-0.1.20170824/lib/qt-5.10/qml/GSettings.1.0/plugins.qmltypes
QQmlComponent: Component is not ready
file:///private/tmp/nix-build-gsettings-qt-0.1.20170824.drv-0/bzr-export/typelist.qml:3:1: plugin cannot be loaded for module "GSettings": Cannot load library /private/tmp/nix-build-gsettings-qt-0.1.20170824.drv-0/bzr-export/GSettings/libGSettingsQmlPlugin.dylib: (dlopen(/private/tmp/nix-build-gsettings-qt-0.1.20170824.drv-0/bzr-export/GSettings/libGSettingsQmlPlugin.dylib, 133): Library not loaded: libgsettings-qt.1.dylib
  Referenced from: /private/tmp/nix-build-gsettings-qt-0.1.20170824.drv-0/bzr-export/GSettings/libGSettingsQmlPlugin.dylib
  Reason: image not found)
make[1]: *** [Makefile.gsettings-qt:698: install_qmltypes] Error 3
make[1]: Leaving directory '/private/tmp/nix-build-gsettings-qt-0.1.20170824.drv-0/bzr-export/GSettings'
make: *** [Makefile:115: sub-GSettings-gsettings-qt-pro-install_subtargets-ordered] Error 2
builder for '/nix/store/4qiifwq9gkaqy0838d7c9xfsj5jhpwpn-gsettings-qt-0.1.20170824.drv' failed with exit code 2
error: build of '/nix/store/4qiifwq9gkaqy0838d7c9xfsj5jhpwpn-gsettings-qt-0.1.20170824.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/gsettings-qt/default.nix
+++ b/pkgs/development/libraries/gsettings-qt/default.nix
@@ -1,0 +1,58 @@
+{ stdenv, fetchbzr, pkgconfig, qmake, qtbase, qtdeclarative, glib, gobjectIntrospection }:
+
+stdenv.mkDerivation rec {
+  name = "gsettings-qt-${version}";
+  version = "0.1.20170824";
+
+  src = fetchbzr {
+    url = http://bazaar.launchpad.net/~system-settings-touch/gsettings-qt/trunk;
+    rev = "85";
+    sha256 = "1kcw0fgdyndx9c0dyha11wkj0gi05spdc1adf1609mrinbb4rnyi";
+  };
+
+  nativeBuildInputs = [
+    pkgconfig
+    qmake
+    gobjectIntrospection
+  ];
+
+  buildInputs = [
+    glib
+    qtdeclarative
+  ];
+
+  patchPhase = ''
+    # force ordered build of subdirs
+    sed -i -e "\$aCONFIG += ordered" gsettings-qt.pro
+
+    # It seems that there is a bug in qtdeclarative: qmlplugindump fails
+    # because it can not find or load the Qt platform plugin "minimal".
+    # A workaround is to set QT_PLUGIN_PATH and QML2_IMPORT_PATH explicitly.
+    export QT_PLUGIN_PATH=${qtbase.bin}/${qtbase.qtPluginPrefix}
+    export QML2_IMPORT_PATH=${qtdeclarative.bin}/${qtbase.qtQmlPrefix}
+
+    substituteInPlace GSettings/gsettings-qt.pro \
+      --replace '$$[QT_INSTALL_QML]' "$out/$qtQmlPrefix" \
+      --replace '$$[QT_INSTALL_BINS]/qmlplugindump' "qmlplugindump"
+
+    substituteInPlace src/gsettings-qt.pro \
+      --replace '$$[QT_INSTALL_LIBS]' "$out/lib" \
+      --replace '$$[QT_INSTALL_HEADERS]' "$out/include"
+  '';
+
+  preInstall = ''
+    # do not install tests
+    for f in tests/Makefile{,.cpptest}; do
+      substituteInPlace $f \
+        --replace "install: install_target" "install: "
+    done
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Qt/QML bindings for GSettings";
+    homepage = https://launchpad.net/gsettings-qt;
+    license = licenses.lgpl3;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.romildo ];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -8986,6 +8986,8 @@ with pkgs;
 
   grpc = callPackage ../development/libraries/grpc { };
 
+  gsettings-qt = libsForQt5.callPackage ../development/libraries/gsettings-qt { };
+
   gst_all_1 = recurseIntoAttrs(callPackage ../development/libraries/gstreamer {
     callPackage = pkgs.newScope (pkgs // { libav = pkgs.ffmpeg; });
   });

